src/main/java/com/myunidays/TrackingUriBuilder.java

```diff
@@ -57,30 +40,58 @@ TrackingUriBuilder prepend(String string) {
+
+        builder.append("&OrderTotal=");
+        if (directTrackingDetails.getOrderTotal() != null)
+            appendDecimal(directTrackingDetails.getOrderTotal());
+
+        builder.append("&ItemsUNiDAYSDiscount=");
+        if (directTrackingDetails.getItemsUnidaysDiscount() != null)
+            appendDecimal(directTrackingDetails.getItemsUnidaysDiscount());
+
+        builder.append("&Code=");
+        if (directTrackingDetails.getCode() != null)
+            builder.append(encode(directTrackingDetails.getCode()));
+
+        builder.append("&ItemsTax=");
+        if (directTrackingDetails.getItemsTax() != null)
+            appendDecimal(directTrackingDetails.getItemsTax());
+
+        builder.append("&ShippingGross=");
+        if (directTrackingDetails.getShippingGross() != null)
+            appendDecimal(directTrackingDetails.getShippingGross());
+
+        builder.append("&ShippingDiscount=");
+        if (directTrackingDetails.getShippingDiscount() != null)
+            appendDecimal(directTrackingDetails.getShippingDiscount());
 
-        if (directTrackingDetails.getCode() != null && !directTrackingDetails.getCode().isEmpty())
-            builder.append("&Code=").append(encode(directTrackingDetails.getCode()));
+        builder.append("&ItemsGross=");
+        if (directTrackingDetails.getItemsGross() != null)
+            appendDecimal(directTrackingDetails.getItemsGross());
 
-        appendDecimalParameter("OrderTotal", directTrackingDetails.getOrderTotal());
-        appendDecimalParameter("ItemsUNiDAYSDiscount", directTrackingDetails.getItemsUnidaysDiscount());
-        appendDecimalParameter("ItemsTax", directTrackingDetails.getItemsTax());
-        appendDecimalParameter("ShippingGross", directTrackingDetails.getShippingGross());
-        appendDecimalParameter("ShippingDiscount", directTrackingDetails.getShippingDiscount());
-        appendDecimalParameter("ItemsGross", directTrackingDetails.getItemsGross());
-        appendDecimalParameter("ItemsOtherDiscount", directTrackingDetails.getItemsOtherDiscount());
-        appendDecimalParameter("UNiDAYSDiscountPercentage", directTrackingDetails.getUnidaysDiscountPercentage());
+        builder.append("&ItemsOtherDiscount=");
+        if (directTrackingDetails.getItemsOtherDiscount() != null)
+            appendDecimal(directTrackingDetails.getItemsOtherDiscount());
 
-        if (directTrackingDetails.getNewCustomer() != null)
-            builder.append("&NewCustomer=").append(directTrackingDetails.getNewCustomer());
+        builder.append("&UNiDAYSDiscountPercentage=");
+        if (directTrackingDetails.getUnidaysDiscountPercentage() != null)
+            appendDecimal(directTrackingDetails.getUnidaysDiscountPercentage());
+
+        builder.append("&NewCustomer=");
+        if (directTrackingDetails.getNewCustomer() != null && directTrackingDetails.getNewCustomer())
+           builder.append("True");

```

Comment:
@jimgolfgti :
This has come up on the other end points -
for V1.1 and V1.0 there are three possible values for NewCustomer - True (is a new customer), False(returning customer) and null (unknown). This is why we have a method to check true, false or null otherwise you never get the false outcome. ( refers to your change in commit 69392c1 )